lib/active_admin/csv_builder.rb

```diff
@@ -44,10 +44,9 @@ def column(name, options = {}, &block)
     def build(controller, csv)
       @collection  = controller.send :find_collection, except: :pagination
       columns      = exec_columns controller.view_context
-      options      = ActiveAdmin.application.csv_options.merge self.options
-      bom          = options.delete :byte_order_mark
-      column_names = options.delete(:column_names) { true }
-      csv_options  = options.except :encoding_options, :humanize_name
```

Comment:
I think leaving the above three lines untouched and use the accessor should work and makes the fix more clear (move the csv option merge to the initializer because it's needed before `build` is called).